src/java/org/apache/cassandra/service/accord/txn/TxnData.java

```diff
@@ -40,18 +41,77 @@
  * Fairly generic holder for result values for Accord txns as well as data exchange during Accord txn execution
  * when read results are returned to the coordinator to compute query results and writes.
  */
-public class TxnData extends HashMap<TxnDataName, TxnDataValue> implements TxnResult, Data
+public class TxnData extends Int2ObjectHashMap<TxnDataValue> implements TxnResult, Data
 {
     private static final long EMPTY_SIZE = ObjectSizes.measure(new TxnData());
 
+    private static final int TXN_DATA_NAME_INDEX_BITS = 32 - 6;
+    private static final int TXN_DATA_NAME_INDEX_MASK = ~(~0 << TXN_DATA_NAME_INDEX_BITS);
```

Comment:
I've never seen this construction before, I like it. I usually do `(0xffffffff >>> (32 - TXN_DATA_NAME_INDEX_BITS))` but this is also neat, particularly the `~0`.